docs/manual/scala/guide/broker/MessageBrokerApi.md

```diff
@@ -0,0 +1,89 @@
+
+Finally, subscribers are grouped together via [`Subscriber.withGroupId`](api/com/lightbend/lagom/scaladsl/api/broker/Subscriber.html#withGroupId\(groupId:String\):com.lightbend.lagom.scaladsl.api.broker.Subscriber[Message]). A subscriber group allows many nodes in your cluster to consume a message stream while ensuring that each message is only handled once by each node in your cluster.  Without subscriber groups, all of your nodes for a particular service would get every message in the stream, leading to their processing being duplicated.  By default, Lagom will use a group id that has the same name as the service consuming the topic.
+
+// TODO: From this point on...
+
+## Polymorphic event streams
+
+Typically you will want to publish more than one type of event to a particular topic. This can be done by creating an interface that each event implements, and then making the events implement that. In order to successfully serialize these events to and from JSON, you will have to include some extra information on your JSON representation of the data.
+
+For example, consider a situation where you have a blog post created event and a blog post published event. Here's what your event structure might look like:
+
+@[content](code/docs/mb/BlogPostService.scala)
+
+And that's how your Play Json formatters could look like:
+
+@[content-formatters](code/docs/mb/BlogPostService.scala)
+
+You will have to implement a custom Message Serializer that adds extra information on each JSON message so that you know what deserializer to use on the other end. The resulting JSON for the `BlogPostCreated` event will look like this:
+
+```json
+{
+ "postId": "23",
+ "title": "new post",
+ "event_type": "postCreated"
+}
+```
+
+While the JSON for the `BlogPostPublished` event will look like this:
+
+```json
+{
+ "postId": "23",
+ "event_type": "postPublished"
+}
+```
+
+You can do that using [Json transformers](https://www.playframework.com/documentation/2.5.x/ScalaJsonTransformers#Case-5:-Put-a-given-value-in-a-new-branch) provided by Play Json: 
+
+@[polymorphic-play-json](code/docs/mb/BlogPostService.scala)
+
+
+This approach has an added maintenance cost. Fortunately there are libraries that expand Play Json features and provide support for algebraic data type serialization. For example: [Play JSON Derived Codecs](https://github.com/julienrf/play-json-derived-codecs).
```

Comment:
@jroper please review this last sentence. I think you mentioned there were others?